src/validator.spec.ts

```diff
@@ -1,82 +1,220 @@
-      succeeded: false,
-      message: 'other message',
-      type: '',
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month and days earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 24);
+      const date = new Date(2018, 11, 30);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month, days and hours earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 30, 10);
+      const date = new Date(2018, 11, 30, 15);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month, days, hours and minutes earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 24, 10, 15);
+      const date = new Date(2018, 12, 24, 10, 33);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object year, month, days, hours, minutes and seconds earlier than customArgs date param', () => {
+      const value = new Date(2018, 12, 30, 15, 33, 30);
+      const date = new Date(2018, 12, 30, 15, 33, 45);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
     });
-  });
 
-  it('should overwrite default message when it feeds value and calls to setErrorMessage', () => {
-    // Arrange
-    const value = 'test';
+    it('should return succeeded validation when value is a valid Date object year, month, days, hours, minutes, seconds and milliseconds earlier than customArgs date param', () => {
+      const value = new Date(2018, 12, 30, 15, 33, 45, 2);
+      const date = new Date(2018, 12, 30, 15, 33, 45, 6);
 
-    setErrorMessage('other message');
+      const result = validator({ value, customArgs: { date } });
 
-    // Act
-    const result = validator({ value });
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return failed validation when value is a valid Date object later than customArgs date param', () => {
+      const value = new Date(2019, 11, 24, 10, 33, 30, 0);
```

Comment:
Do we need more failed specs?

Reply:
Added more failed specs => c37e79cbf2c8aa224995a9d937d0a376120e1a4f